buildtest/cli/report.py

```diff
@@ -600,6 +600,15 @@ def get_buildspecs(self):
         """Return a list of buildspecs in report file"""
         return self.filtered_buildspecs
 
+    def get_failed_tests(self):
+        """Return a list of failed test names from report file"""
+        test_names = []
+        for buildspec in self.filtered_buildspecs:
+            for name in self.report[buildspec].keys():
+                if self.report[buildspec][name][0]["state"] == "FAIL":
```

Comment:
according to this logic we are checking the first test in this case index 0 means the oldest test to see if it fails it is likely that test can have different state if it was run and then fixed so i am not sure if we should just use `0` index or  iterate over entire list and see if any test has a `FAIL` then add to this list

Take for instance this report of failures

```
(buildtest)  ~/Documents/github/buildtest/ [devel] buildtest rt --failure --format name,state
                                                                                  Report File: /Users/siddiq90/Documents/github/buildtest/var/report.json                                                                                  
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ name                                                                                                                                                                                  ┃ state                                           ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│ returncode_list_mismatch                                                                                                                                                              │ FAIL                                            │
├───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┼─────────────────────────────────────────────────┤
│ returncode_list_mismatch                                                                                                                                                              │ FAIL                                            │
├───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┼─────────────────────────────────────────────────┤
│ exit1_fail                                                                                                                                                                            │ FAIL                                            │
├───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┼─────────────────────────────────────────────────┤
│ exit1_fail                                                                                                                                                                            │ FAIL                                            │
└───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┴─────────────────────────────────────────────────┘
```

Take note of the structure of report file

```
  "/Users/siddiq90/Documents/github/buildtest/tutorials/test_status/pass_returncode.yml": {
    "exit1_pass": [
      {
        "id": "44adb1c6",
        "full_id": "44adb1c6-93e6-4916-a014-b15ac3633dad",
        "description": "report exit 1 as PASS",
        "schemafile": "script.schema.json",
        "executor": "generic.local.bash",
        "compiler": null,
        "hostname": "DOE-7086392",
        "user": "siddiq90",
        "testroot": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_pass/44adb1c6",
        "testpath": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_pass/44adb1c6/exit1_pass.sh",
        "stagedir": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_pass/44adb1c6/stage",
        "command": "bash --norc --noprofile -eo pipefail exit1_pass_build.sh",
        "outfile": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_pass/44adb1c6/exit1_pass.out",
        "errfile": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_pass/44adb1c6/exit1_pass.err",
        "buildenv": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_pass/44adb1c6/build-env.txt",
        "buildspec_content": "buildspecs:\n\n  exit1_fail:\n    executor: generic.local.bash\n    type: script\n    description: exit 1 by default is FAIL\n    tags: [tutorials, fail]\n    run: exit 1\n\n  exit1_pass:\n    executor: generic.local.bash\n    type: script\n    description: report exit 1 as PASS\n    run: exit 1\n    tags: [tutorials, pass]\n    status:\n      returncode: [1]\n\n  returncode_list_mismatch:\n    executor: generic.local.bash\n    type: script\n    description: exit 2 failed since it failed to match returncode 1\n    run: exit 2\n    tags: [tutorials, fail]\n    status:\n      returncode: [1, 3]\n\n  returncode_int_match:\n    executor: generic.local.bash\n    type: script\n    description: exit 128 matches returncode 128\n    run: exit 128\n    tags: [tutorials, pass]\n    status:\n      returncode: 128\n",
        "test_content": "#!/bin/bash \n# Content of run section\nexit 1",
        "buildscript_content": "#!/bin/bash\nexport BUILDTEST_TEST_NAME=exit1_pass\nexport BUILDTEST_TEST_ROOT=/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_pass/44adb1c6\nexport BUILDTEST_BUILDSPEC_DIR=/Users/siddiq90/Documents/github/buildtest/tutorials/test_status\nexport BUILDTEST_STAGE_DIR=/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_pass/44adb1c6/stage\n# source executor startup script\nsource /Users/siddiq90/Documents/github/buildtest/var/executor/generic.local.bash/before_script.sh\n# Run generated script\n/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_pass/44adb1c6/stage/exit1_pass.sh\n# Get return code\nreturncode=$?\n# Exit with return code\nexit $returncode",
        "logpath": "/Users/siddiq90/Documents/github/buildtest/var/logs/buildtest_4k23ig1n.log",
        "metrics": {},
        "check": {
          "returncode": true,
          "regex": false,
          "runtime": false
        },
        "tags": "tutorials pass",
        "starttime": "2022/06/22 15:21:49",
        "endtime": "2022/06/22 15:21:49",
        "runtime": "0.092812",
        "state": "PASS",
        "returncode": "1",
        "output": "",
        "error": "",
        "job": {},
        "build_script": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_pass/44adb1c6/exit1_pass_build.sh"
      },
      {
        "id": "b9da2e73",
        "full_id": "b9da2e73-4613-43b2-8a85-b3e295ae32ae",
        "description": "report exit 1 as PASS",
        "schemafile": "script.schema.json",
        "executor": "generic.local.bash",
        "compiler": null,
        "hostname": "DOE-7086392",
        "user": "siddiq90",
        "testroot": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_pass/b9da2e73",
        "testpath": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_pass/b9da2e73/exit1_pass.sh",
        "stagedir": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_pass/b9da2e73/stage",
        "command": "bash --norc --noprofile -eo pipefail exit1_pass_build.sh",
        "outfile": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_pass/b9da2e73/exit1_pass.out",
        "errfile": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_pass/b9da2e73/exit1_pass.err",
        "buildenv": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_pass/b9da2e73/build-env.txt",
        "buildspec_content": "buildspecs:\n\n  exit1_fail:\n    executor: generic.local.bash\n    type: script\n    description: exit 1 by default is FAIL\n    tags: [tutorials, fail]\n    run: exit 1\n\n  exit1_pass:\n    executor: generic.local.bash\n    type: script\n    description: report exit 1 as PASS\n    run: exit 1\n    tags: [tutorials, pass]\n    status:\n      returncode: [1]\n\n  returncode_list_mismatch:\n    executor: generic.local.bash\n    type: script\n    description: exit 2 failed since it failed to match returncode 1\n    run: exit 2\n    tags: [tutorials, fail]\n    status:\n      returncode: [1, 3]\n\n  returncode_int_match:\n    executor: generic.local.bash\n    type: script\n    description: exit 128 matches returncode 128\n    run: exit 128\n    tags: [tutorials, pass]\n    status:\n      returncode: 128\n",
        "test_content": "#!/bin/bash \n# Content of run section\nexit 1",
        "buildscript_content": "#!/bin/bash\nexport BUILDTEST_TEST_NAME=exit1_pass\nexport BUILDTEST_TEST_ROOT=/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_pass/b9da2e73\nexport BUILDTEST_BUILDSPEC_DIR=/Users/siddiq90/Documents/github/buildtest/tutorials/test_status\nexport BUILDTEST_STAGE_DIR=/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_pass/b9da2e73/stage\n# source executor startup script\nsource /Users/siddiq90/Documents/github/buildtest/var/executor/generic.local.bash/before_script.sh\n# Run generated script\n/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_pass/b9da2e73/stage/exit1_pass.sh\n# Get return code\nreturncode=$?\n# Exit with return code\nexit $returncode",
        "logpath": "/Users/siddiq90/Documents/github/buildtest/var/logs/buildtest_woobgm0s.log",
        "metrics": {},
        "check": {
          "returncode": true,
          "regex": false,
          "runtime": false
        },
        "tags": "tutorials pass",
        "starttime": "2022/06/22 15:22:12",
        "endtime": "2022/06/22 15:22:12",
        "runtime": "0.094501",
        "state": "PASS",
        "returncode": "1",
        "output": "",
        "error": "",
        "job": {},
        "build_script": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_pass/b9da2e73/exit1_pass_build.sh"
      }
    ],
    "returncode_list_mismatch": [
      {
        "id": "a6be5158",
        "full_id": "a6be5158-936e-433c-b2a6-a6a458af356d",
        "description": "exit 2 failed since it failed to match returncode 1",
        "schemafile": "script.schema.json",
        "executor": "generic.local.bash",
        "compiler": null,
        "hostname": "DOE-7086392",
        "user": "siddiq90",
        "testroot": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_list_mismatch/a6be5158",
        "testpath": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_list_mismatch/a6be5158/returncode_list_mismatch.sh",
        "stagedir": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_list_mismatch/a6be5158/stage",
        "command": "bash --norc --noprofile -eo pipefail returncode_list_mismatch_build.sh",
        "outfile": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_list_mismatch/a6be5158/returncode_list_mismatch.out",
        "errfile": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_list_mismatch/a6be5158/returncode_list_mismatch.err",
        "buildenv": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_list_mismatch/a6be5158/build-env.txt",
        "buildspec_content": "buildspecs:\n\n  exit1_fail:\n    executor: generic.local.bash\n    type: script\n    description: exit 1 by default is FAIL\n    tags: [tutorials, fail]\n    run: exit 1\n\n  exit1_pass:\n    executor: generic.local.bash\n    type: script\n    description: report exit 1 as PASS\n    run: exit 1\n    tags: [tutorials, pass]\n    status:\n      returncode: [1]\n\n  returncode_list_mismatch:\n    executor: generic.local.bash\n    type: script\n    description: exit 2 failed since it failed to match returncode 1\n    run: exit 2\n    tags: [tutorials, fail]\n    status:\n      returncode: [1, 3]\n\n  returncode_int_match:\n    executor: generic.local.bash\n    type: script\n    description: exit 128 matches returncode 128\n    run: exit 128\n    tags: [tutorials, pass]\n    status:\n      returncode: 128\n",
        "test_content": "#!/bin/bash \n# Content of run section\nexit 2",
        "buildscript_content": "#!/bin/bash\nexport BUILDTEST_TEST_NAME=returncode_list_mismatch\nexport BUILDTEST_TEST_ROOT=/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_list_mismatch/a6be5158\nexport BUILDTEST_BUILDSPEC_DIR=/Users/siddiq90/Documents/github/buildtest/tutorials/test_status\nexport BUILDTEST_STAGE_DIR=/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_list_mismatch/a6be5158/stage\n# source executor startup script\nsource /Users/siddiq90/Documents/github/buildtest/var/executor/generic.local.bash/before_script.sh\n# Run generated script\n/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_list_mismatch/a6be5158/stage/returncode_list_mismatch.sh\n# Get return code\nreturncode=$?\n# Exit with return code\nexit $returncode",
        "logpath": "/Users/siddiq90/Documents/github/buildtest/var/logs/buildtest_4k23ig1n.log",
        "metrics": {},
        "check": {
          "returncode": false,
          "regex": false,
          "runtime": false
        },
        "tags": "tutorials fail",
        "starttime": "2022/06/22 15:21:49",
        "endtime": "2022/06/22 15:21:49",
        "runtime": "0.123639",
        "state": "FAIL",
        "returncode": "2",
        "output": "",
        "error": "",
        "job": {},
        "build_script": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_list_mismatch/a6be5158/returncode_list_mismatch_build.sh"
      },
      {
        "id": "ce18128a",
        "full_id": "ce18128a-f520-489f-8bbe-1d367acd1e25",
        "description": "exit 2 failed since it failed to match returncode 1",
        "schemafile": "script.schema.json",
        "executor": "generic.local.bash",
        "compiler": null,
        "hostname": "DOE-7086392",
        "user": "siddiq90",
        "testroot": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_list_mismatch/ce18128a",
        "testpath": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_list_mismatch/ce18128a/returncode_list_mismatch.sh",
        "stagedir": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_list_mismatch/ce18128a/stage",
        "command": "bash --norc --noprofile -eo pipefail returncode_list_mismatch_build.sh",
        "outfile": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_list_mismatch/ce18128a/returncode_list_mismatch.out",
        "errfile": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_list_mismatch/ce18128a/returncode_list_mismatch.err",
        "buildenv": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_list_mismatch/ce18128a/build-env.txt",
        "buildspec_content": "buildspecs:\n\n  exit1_fail:\n    executor: generic.local.bash\n    type: script\n    description: exit 1 by default is FAIL\n    tags: [tutorials, fail]\n    run: exit 1\n\n  exit1_pass:\n    executor: generic.local.bash\n    type: script\n    description: report exit 1 as PASS\n    run: exit 1\n    tags: [tutorials, pass]\n    status:\n      returncode: [1]\n\n  returncode_list_mismatch:\n    executor: generic.local.bash\n    type: script\n    description: exit 2 failed since it failed to match returncode 1\n    run: exit 2\n    tags: [tutorials, fail]\n    status:\n      returncode: [1, 3]\n\n  returncode_int_match:\n    executor: generic.local.bash\n    type: script\n    description: exit 128 matches returncode 128\n    run: exit 128\n    tags: [tutorials, pass]\n    status:\n      returncode: 128\n",
        "test_content": "#!/bin/bash \n# Content of run section\nexit 2",
        "buildscript_content": "#!/bin/bash\nexport BUILDTEST_TEST_NAME=returncode_list_mismatch\nexport BUILDTEST_TEST_ROOT=/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_list_mismatch/ce18128a\nexport BUILDTEST_BUILDSPEC_DIR=/Users/siddiq90/Documents/github/buildtest/tutorials/test_status\nexport BUILDTEST_STAGE_DIR=/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_list_mismatch/ce18128a/stage\n# source executor startup script\nsource /Users/siddiq90/Documents/github/buildtest/var/executor/generic.local.bash/before_script.sh\n# Run generated script\n/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_list_mismatch/ce18128a/stage/returncode_list_mismatch.sh\n# Get return code\nreturncode=$?\n# Exit with return code\nexit $returncode",
        "logpath": "/Users/siddiq90/Documents/github/buildtest/var/logs/buildtest_woobgm0s.log",
        "metrics": {},
        "check": {
          "returncode": false,
          "regex": false,
          "runtime": false
        },
        "tags": "tutorials fail",
        "starttime": "2022/06/22 15:22:12",
        "endtime": "2022/06/22 15:22:12",
        "runtime": "0.081405",
        "state": "FAIL",
        "returncode": "2",
        "output": "",
        "error": "",
        "job": {},
        "build_script": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_list_mismatch/ce18128a/returncode_list_mismatch_build.sh"
      }
    ],
    "returncode_int_match": [
      {
        "id": "d7d42b2a",
        "full_id": "d7d42b2a-0a2e-4665-9a37-698c26c162c8",
        "description": "exit 128 matches returncode 128",
        "schemafile": "script.schema.json",
        "executor": "generic.local.bash",
        "compiler": null,
        "hostname": "DOE-7086392",
        "user": "siddiq90",
        "testroot": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_int_match/d7d42b2a",
        "testpath": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_int_match/d7d42b2a/returncode_int_match.sh",
        "stagedir": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_int_match/d7d42b2a/stage",
        "command": "bash --norc --noprofile -eo pipefail returncode_int_match_build.sh",
        "outfile": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_int_match/d7d42b2a/returncode_int_match.out",
        "errfile": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_int_match/d7d42b2a/returncode_int_match.err",
        "buildenv": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_int_match/d7d42b2a/build-env.txt",
        "buildspec_content": "buildspecs:\n\n  exit1_fail:\n    executor: generic.local.bash\n    type: script\n    description: exit 1 by default is FAIL\n    tags: [tutorials, fail]\n    run: exit 1\n\n  exit1_pass:\n    executor: generic.local.bash\n    type: script\n    description: report exit 1 as PASS\n    run: exit 1\n    tags: [tutorials, pass]\n    status:\n      returncode: [1]\n\n  returncode_list_mismatch:\n    executor: generic.local.bash\n    type: script\n    description: exit 2 failed since it failed to match returncode 1\n    run: exit 2\n    tags: [tutorials, fail]\n    status:\n      returncode: [1, 3]\n\n  returncode_int_match:\n    executor: generic.local.bash\n    type: script\n    description: exit 128 matches returncode 128\n    run: exit 128\n    tags: [tutorials, pass]\n    status:\n      returncode: 128\n",
        "test_content": "#!/bin/bash \n# Content of run section\nexit 128",
        "buildscript_content": "#!/bin/bash\nexport BUILDTEST_TEST_NAME=returncode_int_match\nexport BUILDTEST_TEST_ROOT=/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_int_match/d7d42b2a\nexport BUILDTEST_BUILDSPEC_DIR=/Users/siddiq90/Documents/github/buildtest/tutorials/test_status\nexport BUILDTEST_STAGE_DIR=/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_int_match/d7d42b2a/stage\n# source executor startup script\nsource /Users/siddiq90/Documents/github/buildtest/var/executor/generic.local.bash/before_script.sh\n# Run generated script\n/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_int_match/d7d42b2a/stage/returncode_int_match.sh\n# Get return code\nreturncode=$?\n# Exit with return code\nexit $returncode",
        "logpath": "/Users/siddiq90/Documents/github/buildtest/var/logs/buildtest_4k23ig1n.log",
        "metrics": {},
        "check": {
          "returncode": true,
          "regex": false,
          "runtime": false
        },
        "tags": "tutorials pass",
        "starttime": "2022/06/22 15:21:49",
        "endtime": "2022/06/22 15:21:49",
        "runtime": "0.118951",
        "state": "PASS",
        "returncode": "128",
        "output": "",
        "error": "",
        "job": {},
        "build_script": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_int_match/d7d42b2a/returncode_int_match_build.sh"
      },
      {
        "id": "f32af797",
        "full_id": "f32af797-7c4a-43ad-a504-a36d06c2179e",
        "description": "exit 128 matches returncode 128",
        "schemafile": "script.schema.json",
        "executor": "generic.local.bash",
        "compiler": null,
        "hostname": "DOE-7086392",
        "user": "siddiq90",
        "testroot": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_int_match/f32af797",
        "testpath": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_int_match/f32af797/returncode_int_match.sh",
        "stagedir": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_int_match/f32af797/stage",
        "command": "bash --norc --noprofile -eo pipefail returncode_int_match_build.sh",
        "outfile": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_int_match/f32af797/returncode_int_match.out",
        "errfile": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_int_match/f32af797/returncode_int_match.err",
        "buildenv": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_int_match/f32af797/build-env.txt",
        "buildspec_content": "buildspecs:\n\n  exit1_fail:\n    executor: generic.local.bash\n    type: script\n    description: exit 1 by default is FAIL\n    tags: [tutorials, fail]\n    run: exit 1\n\n  exit1_pass:\n    executor: generic.local.bash\n    type: script\n    description: report exit 1 as PASS\n    run: exit 1\n    tags: [tutorials, pass]\n    status:\n      returncode: [1]\n\n  returncode_list_mismatch:\n    executor: generic.local.bash\n    type: script\n    description: exit 2 failed since it failed to match returncode 1\n    run: exit 2\n    tags: [tutorials, fail]\n    status:\n      returncode: [1, 3]\n\n  returncode_int_match:\n    executor: generic.local.bash\n    type: script\n    description: exit 128 matches returncode 128\n    run: exit 128\n    tags: [tutorials, pass]\n    status:\n      returncode: 128\n",
        "test_content": "#!/bin/bash \n# Content of run section\nexit 128",
        "buildscript_content": "#!/bin/bash\nexport BUILDTEST_TEST_NAME=returncode_int_match\nexport BUILDTEST_TEST_ROOT=/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_int_match/f32af797\nexport BUILDTEST_BUILDSPEC_DIR=/Users/siddiq90/Documents/github/buildtest/tutorials/test_status\nexport BUILDTEST_STAGE_DIR=/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_int_match/f32af797/stage\n# source executor startup script\nsource /Users/siddiq90/Documents/github/buildtest/var/executor/generic.local.bash/before_script.sh\n# Run generated script\n/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_int_match/f32af797/stage/returncode_int_match.sh\n# Get return code\nreturncode=$?\n# Exit with return code\nexit $returncode",
        "logpath": "/Users/siddiq90/Documents/github/buildtest/var/logs/buildtest_woobgm0s.log",
        "metrics": {},
        "check": {
          "returncode": true,
          "regex": false,
          "runtime": false
        },
        "tags": "tutorials pass",
        "starttime": "2022/06/22 15:22:12",
        "endtime": "2022/06/22 15:22:12",
        "runtime": "0.090261",
        "state": "PASS",
        "returncode": "128",
        "output": "",
        "error": "",
        "job": {},
        "build_script": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/returncode_int_match/f32af797/returncode_int_match_build.sh"
      }
    ],
    "exit1_fail": [
      {
        "id": "4f822edf",
        "full_id": "4f822edf-ebd5-4bbb-811b-3e402ba54222",
        "description": "exit 1 by default is FAIL",
        "schemafile": "script.schema.json",
        "executor": "generic.local.bash",
        "compiler": null,
        "hostname": "DOE-7086392",
        "user": "siddiq90",
        "testroot": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_fail/4f822edf",
        "testpath": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_fail/4f822edf/exit1_fail.sh",
        "stagedir": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_fail/4f822edf/stage",
        "command": "bash --norc --noprofile -eo pipefail exit1_fail_build.sh",
        "outfile": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_fail/4f822edf/exit1_fail.out",
        "errfile": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_fail/4f822edf/exit1_fail.err",
        "buildenv": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_fail/4f822edf/build-env.txt",
        "buildspec_content": "buildspecs:\n\n  exit1_fail:\n    executor: generic.local.bash\n    type: script\n    description: exit 1 by default is FAIL\n    tags: [tutorials, fail]\n    run: exit 1\n\n  exit1_pass:\n    executor: generic.local.bash\n    type: script\n    description: report exit 1 as PASS\n    run: exit 1\n    tags: [tutorials, pass]\n    status:\n      returncode: [1]\n\n  returncode_list_mismatch:\n    executor: generic.local.bash\n    type: script\n    description: exit 2 failed since it failed to match returncode 1\n    run: exit 2\n    tags: [tutorials, fail]\n    status:\n      returncode: [1, 3]\n\n  returncode_int_match:\n    executor: generic.local.bash\n    type: script\n    description: exit 128 matches returncode 128\n    run: exit 128\n    tags: [tutorials, pass]\n    status:\n      returncode: 128\n",
        "test_content": "#!/bin/bash \n# Content of run section\nexit 1",
        "buildscript_content": "#!/bin/bash\nexport BUILDTEST_TEST_NAME=exit1_fail\nexport BUILDTEST_TEST_ROOT=/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_fail/4f822edf\nexport BUILDTEST_BUILDSPEC_DIR=/Users/siddiq90/Documents/github/buildtest/tutorials/test_status\nexport BUILDTEST_STAGE_DIR=/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_fail/4f822edf/stage\n# source executor startup script\nsource /Users/siddiq90/Documents/github/buildtest/var/executor/generic.local.bash/before_script.sh\n# Run generated script\n/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_fail/4f822edf/stage/exit1_fail.sh\n# Get return code\nreturncode=$?\n# Exit with return code\nexit $returncode",
        "logpath": "/Users/siddiq90/Documents/github/buildtest/var/logs/buildtest_4k23ig1n.log",
        "metrics": {},
        "check": {
          "returncode": "N/A",
          "regex": "N/A",
          "runtime": "N/A"
        },
        "tags": "tutorials fail",
        "starttime": "2022/06/22 15:21:49",
        "endtime": "2022/06/22 15:21:49",
        "runtime": "0.11804",
        "state": "FAIL",
        "returncode": "1",
        "output": "",
        "error": "",
        "job": {},
        "build_script": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_fail/4f822edf/exit1_fail_build.sh"
      },
      {
        "id": "6b5f0d35",
        "full_id": "6b5f0d35-a34e-40a8-8c00-fd4ce89a6a4b",
        "description": "exit 1 by default is FAIL",
        "schemafile": "script.schema.json",
        "executor": "generic.local.bash",
        "compiler": null,
        "hostname": "DOE-7086392",
        "user": "siddiq90",
        "testroot": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_fail/6b5f0d35",
        "testpath": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_fail/6b5f0d35/exit1_fail.sh",
        "stagedir": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_fail/6b5f0d35/stage",
        "command": "bash --norc --noprofile -eo pipefail exit1_fail_build.sh",
        "outfile": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_fail/6b5f0d35/exit1_fail.out",
        "errfile": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_fail/6b5f0d35/exit1_fail.err",
        "buildenv": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_fail/6b5f0d35/build-env.txt",
        "buildspec_content": "buildspecs:\n\n  exit1_fail:\n    executor: generic.local.bash\n    type: script\n    description: exit 1 by default is FAIL\n    tags: [tutorials, fail]\n    run: exit 1\n\n  exit1_pass:\n    executor: generic.local.bash\n    type: script\n    description: report exit 1 as PASS\n    run: exit 1\n    tags: [tutorials, pass]\n    status:\n      returncode: [1]\n\n  returncode_list_mismatch:\n    executor: generic.local.bash\n    type: script\n    description: exit 2 failed since it failed to match returncode 1\n    run: exit 2\n    tags: [tutorials, fail]\n    status:\n      returncode: [1, 3]\n\n  returncode_int_match:\n    executor: generic.local.bash\n    type: script\n    description: exit 128 matches returncode 128\n    run: exit 128\n    tags: [tutorials, pass]\n    status:\n      returncode: 128\n",
        "test_content": "#!/bin/bash \n# Content of run section\nexit 1",
        "buildscript_content": "#!/bin/bash\nexport BUILDTEST_TEST_NAME=exit1_fail\nexport BUILDTEST_TEST_ROOT=/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_fail/6b5f0d35\nexport BUILDTEST_BUILDSPEC_DIR=/Users/siddiq90/Documents/github/buildtest/tutorials/test_status\nexport BUILDTEST_STAGE_DIR=/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_fail/6b5f0d35/stage\n# source executor startup script\nsource /Users/siddiq90/Documents/github/buildtest/var/executor/generic.local.bash/before_script.sh\n# Run generated script\n/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_fail/6b5f0d35/stage/exit1_fail.sh\n# Get return code\nreturncode=$?\n# Exit with return code\nexit $returncode",
        "logpath": "/Users/siddiq90/Documents/github/buildtest/var/logs/buildtest_woobgm0s.log",
        "metrics": {},
        "check": {
          "returncode": "N/A",
          "regex": "N/A",
          "runtime": "N/A"
        },
        "tags": "tutorials fail",
        "starttime": "2022/06/22 15:22:12",
        "endtime": "2022/06/22 15:22:13",
        "runtime": "0.091151",
        "state": "FAIL",
        "returncode": "1",
        "output": "",
        "error": "",
        "job": {},
        "build_script": "/Users/siddiq90/Documents/github/buildtest/var/tests/generic.local.bash/pass_returncode/exit1_fail/6b5f0d35/exit1_fail_build.sh"
      }
    ]
  }

```